File: cortex-spring-boot-autoconfigure/src/main/java/ai/chatur/cortex/spring/dataset/DatasetConfiguration.java
package ai.chatur.cortex.spring.dataset;

import org.apache.jena.query.Dataset;
import org.apache.jena.tdb2.TDB2Factory;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class DatasetConfiguration {

  @Bean
  @Qualifier("assertions")
  Dataset assertions() {
    return TDB2Factory.createDataset();
  }
}

File: src/main/kotlin/ai/chatur/cortex/core/DatasetConfiguration.kt
package ai.chatur.cortex.core

import org.apache.jena.query.Dataset
import org.apache.jena.tdb2.TDB2Factory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class DatasetConfiguration {

  @Bean
  fun dataset(): Dataset {
    return TDB2Factory.createDataset()
  }

  @Bean
  fun datasetService(ds: Dataset): DatasetService {
    return DatasetService(ds)
  }
}
